internal/dpcnisyncerserver/server.go

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright(c) 2023 Intel Corporation.
+ * Copyright(c) Red Hat Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dpcnisyncerserver
+
+import (
+	"context"
+	"net"
+	"os"
+	"time"
+
+	// Replace with the actual proto package
+	"github.com/intel/afxdp-plugins-for-kubernetes/constants"
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/bpf"
+	pb "github.com/intel/afxdp-plugins-for-kubernetes/internal/dpcnisyncer"
+	"github.com/pkg/errors"
+	logging "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+	protocol = "unix"
+)
+
+var (
+	sockAddr = pluginapi.DevicePluginPath + constants.Plugins.DevicePlugin.DevicePrefix + "-" + "syncer.sock"
+)
+
+type SyncerServer struct {
+	pb.UnimplementedNetDevServer
+	mapManagers []bpf.PoolBpfMapManager
+	grpcServer  *grpc.Server
+}
+
+func (s *SyncerServer) RegisterMapManager(b bpf.PoolBpfMapManager) {
+
+	if s.mapManagers != nil {
+		for _, v := range s.mapManagers {
+			if v.Manager.GetName() == b.Manager.GetName() {
+				logging.Infof("%s is already registered", b.Manager.GetName())
+				return
+			}
+		}
+	}
+
+	s.mapManagers = append(s.mapManagers, b)
+}
+
+func (s *SyncerServer) DelNetDev(ctx context.Context, in *pb.DeleteNetDevReq) (*pb.DeleteNetDevResp, error) {
+	netDevName := in.GetName()
```

Comment:
should check if bpfmappinning is configured in case of misconfiguration on the CNI side